File: app/src/main/java/com/gnest/remember/view/fragments/ArchiveItemFragment.java
package com.gnest.remember.view.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.Menu;
import android.view.MenuInflater;

import com.gnest.remember.R;
import com.gnest.remember.presenter.ArchiveFragmentPresenter;
import com.gnest.remember.presenter.IListFragmentPresenter;

import butterknife.BindString;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class ArchiveItemFragment extends ListItemFragment {

    @BindString(R.string.note_unarchived_message)
    String noteUnarchiveMessage;
    @BindString(R.string.note_unarchived_message_1)
    String noteUnarchiveMessage1;
    @BindString(R.string.note_unarchived_message_2)
    String noteUnarchiveMessage2;


    public static ArchiveItemFragment newInstance(int columnCount, int memoSize, int margins) {
        ArchiveItemFragment fragment = new ArchiveItemFragment();
        Bundle args = new Bundle();
        args.putInt(ListItemFragment.ARG_COLUMN_COUNT, columnCount);
        args.putInt(ListItemFragment.ARG_MEMO_SIZE, memoSize);
        args.putInt(ListItemFragment.ARG_MEMO_MARGINS, margins);
        fragment.setArguments(args);
        return fragment;
    }

    @NonNull
    @Override
    public IListFragmentPresenter createPresenter() {
        return new ArchiveFragmentPresenter();
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        //Has to be empty, so it won't inflate its parents menu layout here
    }

    @Override
    String getArchiveActionPluralForm(int plural) {
        switch (plural) {
            case 2:
                return noteUnarchiveMessage2;
            case 1:
                return noteUnarchiveMessage1;
            default:
                return noteUnarchiveMessage;
        }
    }

    public void waitForLoadAndOpenFromNotification(int id) {
        getDataLoadedSubject()
                .subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread())
                .distinctUntilChanged()
                .subscribe(loaded -> openFromNotification(id));
    }
}

File: app/src/main/java/com/gnest/remember/ui/fragments/ArchiveItemFragment.kt
package com.gnest.remember.ui.fragments

class ArchiveItemFragment : ListItemFragment() {

//    override fun createPresenter(): IListFragmentPresenter {
//        return ArchiveFragmentPresenter()
//    }
//
//    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
//        //Has to be empty, so it won't inflate its parents menu layout here
//    }

//    override fun handleOnBackPressed(): Boolean = when {
//        myGridLayoutManager?.orientation == RecyclerView.HORIZONTAL -> super.handleOnBackPressed()
//        else -> {
//            findNavController().navigateUp()
//            true
//        }
//    }
//
//    override fun onOptionsItemSelected(item: MenuItem): Boolean {
//        return when (item.itemId) {
//            android.R.id.home -> {
//                findNavController().navigateUp()
//                true
//            }
//            else -> super.onOptionsItemSelected(item)
//        }
//    }
//
//    override fun onSingleChoiceMemoClicked(memo: Memo) {
//        myGridLayoutManager
//                ?.takeIf { it.orientation == RecyclerView.VERTICAL }
//                ?.let { it.openItem(memo.position) }
//    }
//
//    override fun getArchiveActionPluralForm(plural: Int): String = when (plural) {
//        2 -> getString(R.string.note_unarchived_message_2)
//        1 -> getString(R.string.note_unarchived_message_1)
//        else -> getString(R.string.note_unarchived_message)
//    }
}